### PR TITLE
Logback MDC auto-instrumentation gets muzzled and does not work

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1201

### CODE REVIEW

File: instrumentation/logback/logback-1.0.0/auto/src/main/java/io/opentelemetry/instrumentation/auto/logback/v1_0_0/LoggingEventInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            .and(takesArguments(0)),
+        LoggingEventInstrumentation.class.getName() + "$GetMdcAdvice");
+  }
+
+  public static class GetMdcAdvice {
+    @Advice.OnMethodExit(suppress = Throwable.class)
+    public static void onExit(
+        @Advice.This ILoggingEvent event,
+        @Advice.Return(typing = Typing.DYNAMIC, readOnly = false) Map<String, String> contextData) {
+      if (contextData != null && contextData.containsKey("traceId")) {
```
Comment (mateuszrzeszutek) [inline instrumentation/logback/logback-1.0.0/auto/src/main/java/io/opentelemetry/instrumentation/auto/logback/v1_0_0/LoggingEventInstrumentation.java:81]: I'll replace `"traceId"` and others either here or in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1200 - depending on which one gets merged first.

File: instrumentation/logback/logback-1.0.0/auto/src/main/java/io/opentelemetry/instrumentation/auto/logback/v1_0_0/LoggingEventInstrumentation.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        // Assume already instrumented event if traceId is present.
+        return;
+      }
+
+      Span currentSpan = InstrumentationContext.get(ILoggingEvent.class, Span.class).get(event);
+      if (currentSpan != null && !currentSpan.getContext().isValid()) {
```
Comment (trask) [inline instrumentation/logback/logback-1.0.0/auto/src/main/java/io/opentelemetry/instrumentation/auto/logback/v1_0_0/LoggingEventInstrumentation.java:87]: I think this:
```suggestion
      if (currentSpan == null || !currentSpan.getContext().isValid()) {
```

**Review by trask — COMMENTED** [review]
Thanks 👍

**Review by iNikem — APPROVED** [review]

File: instrumentation/logback/logback-1.0.0/library/src/main/java/io/opentelemetry/instrumentation/logback/v1_0_0/UnionMap.java [hunk 3]
```diff
@@ -29,14 +29,14 @@
  * An immutable view over two maps, with keys resolving from the first map first, or otherwise the
  * second if not present in the first.
  */
-final class UnionMap<K, V> extends AbstractMap<K, V> {
+public final class UnionMap<K, V> extends AbstractMap<K, V> {
```
Comment (anuraaga) [inline instrumentation/logback/logback-1.0.0/library/src/main/java/io/opentelemetry/instrumentation/logback/v1_0_0/UnionMap.java:32]: Can you move this into a subpackage `internal`? We shouldn't add this to our public API

File: instrumentation/logback/logback-1.0.0/auto/src/main/java/io/opentelemetry/instrumentation/auto/logback/v1_0_0/LoggingEventInstrumentation.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply instrumentation/logback/logback-1.0.0/auto/src/main/java/io/opentelemetry/instrumentation/auto/logback/v1_0_0/LoggingEventInstrumentation.java:87]: 🤦  - that's correct, that was a stupid mistake on my side.

File: instrumentation/logback/logback-1.0.0/library/src/main/java/io/opentelemetry/instrumentation/logback/v1_0_0/UnionMap.java (hunk 3 above)
Comment (mateuszrzeszutek) [reply instrumentation/logback/logback-1.0.0/library/src/main/java/io/opentelemetry/instrumentation/logback/v1_0_0/UnionMap.java:32]: Done, thanks!

**Review by anuraaga — APPROVED** [review]

**Pull request merged**